src/Hydrators/GamesHydrator.php

```diff
@@ -2,13 +2,23 @@
 namespace GameListing\Hydrators;
 
 use GameListing\Entities\PCGameEntity;
+use GameListing\Entities\DetailedPCGameEntity;
 
 class GamesHydrator
 {
     public static function getAllGames(\PDO $db): array
     {
-        $query = $db->query('SELECT `title`, `genre`, `thumbnail` FROM `pc-games`;');
+        $query = $db->query('SELECT `title`, `genre`, `thumbnail`, `id` FROM `pc-games`;');
         $query->setFetchMode(\PDO::FETCH_CLASS, PCGameEntity::class);
         return $query->fetchAll();
     }
+
+    public static function getGameById(\PDO $db, int $id): DetailedPCGameEntity
+    {
+        $query = $db->query('SELECT `id`, `title`, `genre`, `thumbnail`, `shortDescription`, `gameUrl`,
+       `genre`, `platform`, `publisher`, `developer`, `releaseDate`, `freetogameProfileUrl`
+        FROM `pc-games` WHERE `id` = ' . $id . ';');
```

Comment:
SQL injection vulnerability 😱

Reply:
updated

Reply:
it's still concatenating, need to use :id in the query

Reply:
should be all good